src/cache_cf.cc

```diff
@@ -3881,25 +3881,30 @@ parsePortCfg(AnyP::PortCfgPointer *head, const char *optionName)
 
     if (s->transport.protocol == AnyP::PROTO_HTTPS) {
         s->secure.encryptTransport = true;
+        const auto hijacked = s->flags.isIntercepted();
 #if USE_OPENSSL
-        /* ssl-bump on https_port configuration requires either tproxy or intercept, and vice versa */
-        const bool hijacked = s->flags.isIntercepted();
-        if (s->flags.tunnelSslBumping && !hijacked) {
-            debugs(3, DBG_CRITICAL, "FATAL: ssl-bump on https_port requires tproxy/intercept which is missing.");
+        /* ssl-bump on https_port configuration requires one of tproxy, intercept or require-proxy-header and vice versa */
+        const auto sslBumpRequiredOption = hijacked || s->flags.proxySurrogate;
+        if (s->flags.tunnelSslBumping && !sslBumpRequiredOption) {
+            debugs(3, DBG_CRITICAL, "FATAL: ssl-bump on https_port requires tproxy/intercept/require-proxy-header which is missing.");
             self_destruct();
             return;
         }
-        if (hijacked && !s->flags.tunnelSslBumping) {
-            debugs(3, DBG_CRITICAL, "FATAL: tproxy/intercept on https_port requires ssl-bump which is missing.");
+        if (sslBumpRequiredOption && !s->flags.tunnelSslBumping) {
+            debugs(3, DBG_CRITICAL, "FATAL: tproxy/intercept/require-proxy-header on https_port requires ssl-bump which is missing.");
             self_destruct();
             return;
         }
 #endif
-        if (s->flags.proxySurrogate) {
-            debugs(3,DBG_CRITICAL, "FATAL: https_port: require-proxy-header option is not supported on HTTPS ports.");
+        if (s->flags.proxySurrogate && (hijacked || s->flags.accelSurrogate)) {
+            debugs(3, DBG_CRITICAL, "FATAL: require-proxy-header option is incompatible with tproxy/intercept/accel for https_port.");
             self_destruct();
             return;
         }
+
+        if (s->flags.tunnelSslBumping && s->flags.proxySurrogate)
+            s->flags.natIntercept = true; // emulate interception for PROXY protocol
```

Comment:
I really do not like this hack. I am sure it breaks some existing natIntercept-using code already, and the number of such bugs will only increase as folks will continue to rely on the now-overloaded natIntercept to mean just NAT interception. Squid have suffered greatly from similar overloading hacks, and we are trying to avoid them with introduction of classes like TrafficMode.

I think we have to fix it for the official version. Adding a TrafficMode::proxyProtocol flag is easy. Fortunately, the number of current natIntercept uses is small, so we can adjust that code as needed without a lot of changes. We may face problems with Connection::flags (e.g., `COMM_INTERCEPTION`). Hopefully, adding `COMM_PROXY_PROTOCOL` is either unnecessary or not difficult. I did not check that because even if it is difficult, I do not see a better alternative.

To keep the v5-branch viable, I would try to fix this in this branch before proceeding with the master/v6 port.